controllers/users.js

```diff
@@ -128,6 +128,30 @@ const getUsers = async (req, res) => {
       }
     }
 
+    // getting user details by discord id if present.
+    const dev = req.query.dev === "true";
+    if (dev) {
+      if (req.query.discordId) {
+        const discordId = req.query.discordId;
```

Comment:
in the line const discordId = req.query.discordId;. The req.query object is a JavaScript object that contains the query parameters passed in the URL. In this case, the query parameter is discordId. However, the code is trying to access the discordId property of the req.query object before it has been initialized. This will cause an error.